### PR TITLE
Add several clean options to the build script

### PR DESCRIPTION
**Pull request by vithom**
* Add a --clean-build option to clean the build directory
* Add a --clean-install option to clean the install directory
* Add a --clean-all option to clean both
* Add their 'force' variants
* Add a log function to the script

Implements #7002

### CODE REVIEW

**Comment by TurboGit** [comment]
I would remove the force variant and use a `-f / --force` option to trigger it.

**Comment by parafin** [comment]
--clean-install option is very dangerous if install directory is changed to some path not unique to darktable, e.g. /usr/local/.

**Comment by johnny-bit** [comment]
As was said in #7002 you NEED to take into account installed files and never remove anything that wasn't clearly auto-created.

in build dir there should be `install_manifest.txt` which lists all installed files. and there should be also cmake_uninstall.cmake which does the cleaning...

**Comment by parafin** [comment]
On the other hand it might be useful to be able to install into really clean install dir to be sure nothing interferes with the installation (e.g. some custom build external modules)... I can propose the following alternative - if install action is trying to copy files into non-empty directory - suggest to remove all files from it (with a big fat warning about other stuff not being installed to the same prefix). Or better yet - compile the said install_manifest.txt during installation somehow, then after the install is finished, compare it to the contents of the prefix and if there are some extra files suggest to remove them (printing the list if not too big). install_manifest.txt then can be used for uninstall action (uninstall is more to the point name for this than clean).

**Comment by vithom** [comment]
Hello,
@TurboGit : I replace the force variants with a -f general option
@johnny-bit : I use the manifest.txt generated by cmake to clean the install directory. Bash script does the same thing than the cmake_uninstall.cmake does.

File: build.sh [hunk 1]
```diff
@@ -22,6 +23,10 @@ BUILD_GENERATOR_DEFAULT="Unix Makefiles"
 BUILD_GENERATOR="$BUILD_GENERATOR_DEFAULT"
 MAKE_TASKS=-1
 ADDRESS_SANITIZER=0
+DO_CLEAN_BUILD=0
+DO_CLEAN_INSTALL=0
+MANIFEST_FILE="$BUILD_DIR/install_manifest.txt"
```
Comment (TurboGit) [inline build.sh:28]: I don't see any update for this file. Always empty? How can this be used to clean install directory?

**Comment by vithom** [comment]
@TurboGit : This file is generated by cmake when the build is done. It's not under version control.

File: build.sh (hunk 1 above)
Comment (vithom) [reply build.sh:28]: This file is generated by cmake when the build is done. It's not under version control.

Comment (TurboGit) [reply build.sh:28]: I don't have this file after a successful build using build.sh.

Comment (vithom) [reply build.sh:28]: Even in the build/ directory when you launch with --install ?

**Comment by TurboGit** [comment]
Too close from the release for this, moved to 3.8.

**Review by TurboGit — APPROVED** [review]
Looks good, thanks !

**Comment by TurboGit** [comment]
Arg!!! Just forgot that you need to add the new option into the `print_help()` routine. Can you do that ? TIA.

**Comment by vithom** [comment]
Yes, no problem. I will do it as soon as I come back from vacation.

**Comment by TurboGit** [comment]
It has been done by @johnny-bit ! So nothing more to do.

**Pull request merged**